bentoml/adapters/multi_image_input.py

```diff
@@ -39,59 +50,96 @@ class MultiImageInput(BaseInputAdapter):
         self.pilmode = pilmode
-        self.accept_image_formats = (
-            accept_image_formats or get_default_accept_image_formats()
+        self.accepted_image_formats = (
+            accepted_image_formats or get_default_accept_image_formats()
         )
 
-    def handle_batch_request(self, requests, func):
-        """Handles an HTTP request, convert it into corresponding data
-        format that user API function is expecting, and return API
-        function result as the HTTP response to client
-
-        :param requests: List of flask request object
-        :param func: user API function
-        """
-        raise NotImplementedError
+    def handle_request(self, request: Request, func):
+        files = {
+            name: self.read_file(file.filename, file.stream)
+            for (name, file) in request.files.items()
+        }
+        result = func((files,))[0]
+        return self.output_adapter.to_response(result, request)
+
+    def read_file(self, name: str, file):
+        safe_name = secure_filename(name)
+        verify_image_format_or_raise(safe_name, self.accepted_image_formats)
+        return imread(file, pilmode=self.pilmode)
+
+    def handle_batch_request(
+        self, requests: Iterable[SimpleRequest], func
+    ) -> Iterable[SimpleResponse]:
+        raise NotImplementedError(
+            "The batch processing architecture does not currently support multipart "
+            "requests, which are required for multi-image requests"
+        )
 
     def handle_cli(self, args, func):
         """Handles an CLI command call, convert CLI arguments into
         corresponding data format that user API function is expecting, and
         prints the API function result to console output
-
         :param args: CLI arguments
         :param func: user API function
         """
-        raise NotImplementedError
+        parser = argparse.ArgumentParser()
+        for input_name in self.input_names:
+            parser.add_argument(input_name, required=True)
+        args, unknown_args = parser.parse_known_args(args)
+        files = {
+            input_name: self.read_file(
+                pathlib.Path(args[input_name]).name, args[input_name]
+            )
+            for input_name in self.input_names
+        }
+        result = func((files,))[0]
+        return self.output_adapter.to_cli(result, unknown_args)
 
     def handle_aws_lambda_event(self, event, func):
         """Handles a Lambda event, convert event dict into corresponding
         data format that user API function is expecting, and use API
         function result as response
-
         :param event: AWS lambda event data of the python `dict` type
         :param func: user API function
         """
-        raise NotImplementedError
+        content_type = event.headers['content-type']
+        if "multipart/form-data" in content_type:
+            files = {}
+
+            request = decoder.MultipartDecoder(event.body, content_type)
```

Comment:
This relies on the content type including the multipart boundary, which should be a safe assumption, but I thought I should note it anyways.